From aad20850ef2b08cb01229270476b4852507de058 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Wed, 05 Feb 2020 10:12:19 +0200
Subject: [PATCH] [SW-8229] support signature bypass fuse

If efuse is clean we dont validate signature, only after
burning this efuse will signature validation occur.

Change-Id: Iac0428277739b2e20a01c2f9d3fa82eced1f96c2
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/efuse.h b/common/efuse.h
index 589d4bc..4391e0e 100644
--- a/common/efuse.h
+++ b/common/efuse.h
@@ -20,6 +20,7 @@
 #define EFUSE_SYSCFG_DISABLE_DEBUG_IF_MASK	BIT(12)
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION	13
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION_MASK	0x3
+#define EFUSE_SYSCFG_ENABLE_VALIDATION_MASK	BIT(10)
 #define EFUSE_SYSCFG_KEY_LOCATION		7
 #define EFUSE_SYSCFG_KEY_LOCATION_MASK		0x3
 #define EFUSE_SYSCFG_REVOKATION_LOCATION	5
diff --git a/common/secureboot.c b/common/secureboot.c
index ca8c9a0..01e66f4 100644
--- a/common/secureboot.c
+++ b/common/secureboot.c
@@ -8,6 +8,7 @@
 #include <stdio.h>
 #include "sb.h"
 #include "arcboot.h"
+#include "efuse.h"
 #include "secureboot.h"
 #include "security.h"
 
@@ -45,6 +46,10 @@
 	unsigned char *test_public_key = test_phase0b_sig -
 					 test_public_key_len;
 	uint32_t fw_version = base_readl((u32)test_phase0b_data);
+	if ((arcboot_rom_global.config_fuse &
+	     EFUSE_SYSCFG_ENABLE_VALIDATION_MASK) == 0) {
+		return SECUREBOOT_VERIFY_SUCCESS;
+	}
 
 	/* check minimal valid security FW version */
 	if (security_check_valid_fw_version(fw_version) !=
